examples/simple-lambda.hs

```diff
@@ -40,3 +45,53 @@ code = "\
 \  callback(null, \"some success message\"); \
 \ } \
 \ "
+
+
+role :: Resource
+role = 
+  resource "IAMRole" $
+  IAMRoleProperties $
+  iamRole rolePolicyDocumentObject
+  & iamrPolicies ?~ [ executePolicy ]
+  & iamrRoleName ?~ "MyLambdaBasicExecutionRole"
+  & iamrPath ?~ "/"
+
+  where
+    executePolicy = 
+      iamPolicies
+      (HM.fromList
```

Comment:
Did the `OverloadedLists` extension not work here? I thought you would be able to create the `Object` literally. 

There are a couple options:

1. An `Object` is simply a type alias for `HashMap`. If I have to create one literally, I use the [`object`](https://hackage.haskell.org/package/aeson-1.0.1.0/docs/Data-Aeson.html#v:object) function from aeson instead of `HashMap.fromList`?
2. I tend to use `aeson-qq` to create the `Object` and then pass that into `iamPolicies`. For this example, I would to this (note, I haven't actually type-checked this :smile:):

```haskell
...
    executePolicy = iamPolicies executePolicyObject "MyLambdaExecutionPolicy"
...

executePolicyObject :: Object
(Object executePolicyObject) = [aesonQQ|
{
   "Version": "2012-10-17",
   "Statement": [
      {
         "Effect": "Allow",
         "Action": [
            "logs:CreateLogGroup",
            "logs:CreateLogStream",
            "logs:PutLogEvents"
         ],
         "Resource": "*"
      }
   ]
}
|]
```